### PR TITLE
#2105 Change spark-jobs config keys prefix from `menas.rest` to `enceladus.rest`

### PR DESCRIPTION
**Pull request by jakipatryk**
As requested in #2105, configuration keys (spark-jobs) with prefix `menas.rest` have been changed to have prefix `enceladus.rest` instead.

Note: this PR only changes these configurations, code refactoring to remove other mentions of Menas when REST API is intended should also be covered, in another issue/PR.

## Release notes suggestion
spark-jobs: Configuration keys starting with `menas.rest` were changed to `enceladus.rest`.

### CODE REVIEW

File: rest-api/src/main/resources/scheduling/oozie/workflow_template.xml [hunk 1]
```diff
@@ -62,7 +62,7 @@
                 --num-executors $confNumExecutors
                 --executor-memory $confExecutorMemory
                 --driver-cores $driverCores
-                --conf $sparkConfQuotesspark.driver.extraJavaOptions=-Dmenas.rest.uri='$menasRestURI' -Dspline.producer.url='$lineageWriteApiUrl' -Dlog4j.configuration='spark-log4j.properties' -Dconformance.mappingtable.pattern='$mappingTablePattern'$sparkConfQuotes $splineModeConf
+                --conf $sparkConfQuotesspark.driver.extraJavaOptions=-Denceladus.rest.uri='$menasRestURI' -Dspline.producer.url='$lineageWriteApiUrl' -Dlog4j.configuration='spark-log4j.properties' -Dconformance.mappingtable.pattern='$mappingTablePattern'$sparkConfQuotes $splineModeConf
```
Comment (dk1844) [inline rest-api/src/main/resources/scheduling/oozie/workflow_template.xml:65]: dtto: Feel free to leave this one for the subsequent PR that solves the field names if you feel it rather belongs there.

```suggestion
                --conf $sparkConfQuotesspark.driver.extraJavaOptions=-Denceladus.rest.uri='$enceladusRestURI' -Dspline.producer.url='$lineageWriteApiUrl' -Dlog4j.configuration='spark-log4j.properties' -Dconformance.mappingtable.pattern='$mappingTablePattern'$sparkConfQuotes $splineModeConf
```

File: rest-api/src/main/resources/scheduling/oozie/workflow_template.xml [hunk 2]
```diff
@@ -32,7 +32,7 @@
                 --num-executors $stdNumExecutors
                 --executor-memory $stdExecutorMemory
                 --driver-cores $driverCores
-                --conf $sparkConfQuotesspark.driver.extraJavaOptions=-Dmenas.rest.uri='$menasRestURI' -Dspline.producer.url='$lineageWriteApiUrl' -Dlog4j.configuration='spark-log4j.properties'$sparkConfQuotes $splineMode
+                --conf $sparkConfQuotesspark.driver.extraJavaOptions=-Denceladus.rest.uri='$menasRestURI' -Dspline.producer.url='$lineageWriteApiUrl' -Dlog4j.configuration='spark-log4j.properties'$sparkConfQuotes $splineMode
```
Comment (dk1844) [inline rest-api/src/main/resources/scheduling/oozie/workflow_template.xml:35]:
```suggestion
                --conf $sparkConfQuotesspark.driver.extraJavaOptions=-Denceladus.rest.uri='$enceladusRestURI' -Dspline.producer.url='$lineageWriteApiUrl' -Dlog4j.configuration='spark-log4j.properties'$sparkConfQuotes $splineMode
```

Feel free to leave this one for the subsequent PR that solves the field names if you feel it rather belongs there.

File: spark-jobs/src/test/resources/application.conf [hunk 3]
```diff
@@ -13,7 +13,7 @@
 
 enceladus.version=${project.version}
 
-menas.rest.uri="http://localhost:8080/menas/api"
+enceladus.rest.uri="http://localhost:8080/menas/api"
```
Comment (dk1844) [inline spark-jobs/src/test/resources/application.conf:16]: This seems to be always overwritten in practice with actual URI when running, but providing a different sensible default/hint (e.g. `http://localhost:8080/rest_api`) would make more sense to me.

**Review by dk1844 — DISMISSED** [review]
LGTM (just read the code)
 - I hope I have not missed any unresolved config key usages

**Review by dk1844 — APPROVED** [review]

**Review by Zejnilovic — APPROVED** [review]

**Review by benedeki — APPROVED** [review]

**Comment by benedeki** [comment]
- [X] code reviewed
- [X] pulled
- [X] built
- [X] run

**Pull request merged**